packages/ui/src/molecules/RadioGroup/RadioOption.tsx

```diff
@@ -0,0 +1,55 @@
+import type { InputHTMLAttributes } from 'react'
+import React, { forwardRef } from 'react'
+
+import { Radio } from '../..'
+import { useRadioOption } from './useRadioOption'
+
+export interface RadioOptionProps
+  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'type'> {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress,
+   * testing-library, and jest).
+   */
+  testId?: string
+  /**
+   * Attribute to set aria-label option as required.
+   */
+  label: string
+  /**
+   * Value to be emitted when radio is clicked.
+   */
+  value: string | number
+  /**
+   * Children to show in radio.
+   */
+  children?: React.ReactNode
+}
+
+const RadioOption = forwardRef<HTMLInputElement, RadioOptionProps>(
+  function RadioOption(
+    { label, value, children, testId = 'store-radio-option', ...otherOptions },
+    ref
+  ) {
+    const { name, value: option, onChange } = useRadioOption()
+
+    const checked = value === option
+
+    return (
+      <label aria-label={label} data-store-radio-option>
+        <Radio
+          {...otherOptions}
+          ref={ref}
+          name={name}
+          checked={checked}
+          onChange={() => {
+            onChange?.(value)
+          }}
+          testId={testId}
+        />
+        {children ?? <span>{label}</span>}
```

Comment:
I really don't like this fallback to the `span` with label. I think users can do it by themselves, and we'd not be offering any way to style this `span` also. I think we should accept the label due to the a11y advantages, but that's it.

```suggestion
        {children}
```

Reply:
Okay, i put label inside the span to help in style with 
`
[data-store-radio-option] input[type='radio'] + span {
  @apply text-pink-600;
}
`

but i agree to let users do it by themselves